docs/implementation.md

```diff
@@ -12,9 +12,70 @@
 
 # DID Cache
 
+The DID cache holds most of the state of a Sidetree node. It is a singleton class with the following methods for state update and retrieval.
 
-# Merkle Rooter
+## Record
+
+This is the core method to update state with the following signature.
+
+```javascript
+public record(stateModifyOp: any, blockId: BlockId): URL
+```
+The `stateModifyOp` is a JSON object representing a create, update, or a delete operation. Recall from the protocol description that the SHA256 hash of this object is the *versionURL* of the document version produced as the result of this operation. The `blockId` parameter identifies the blockchain block that this operation is anchored to.
```

Comment:
Does the record method really need to return URL (operation hash) since the caller has the operation in the first place? Is this more of a convenience/sanity return value?

Reply:
Removed return type for now, but if we do, it will probably be a Buffer representing a multihash.